tfe/resource_tfe_variable_set.go

```diff
@@ -0,0 +1,181 @@
+package tfe
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+
+	tfe "github.com/hashicorp/go-tfe"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var variableSetIdRegexp = regexp.MustCompile("varset-[a-zA-Z0-9]{16}$")
+
+func resourceTFEVariableSet() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceTFEVariableSetCreate,
+		Read:   resourceTFEVariableSetRead,
+		Update: resourceTFEVariableSetUpdate,
+		Delete: resourceTFEVariableSetDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"global": {
+				Type:          schema.TypeBool,
+				Optional:      true,
+				Default:       false,
+				ConflictsWith: []string{"workspaces"},
+			},
+
+			"organization": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"workspaces": {
```

Comment:
Rename to `workspace_ids` ?

Reply:
The thinking here is that other references, like `organization`, also take an identifier as an argument, but the label is for the whole object(s) being referenced. But I see other code using worksace_ids, so I will update this to match.